hack/k8s/example-coord-pg.template.yaml

```diff
@@ -0,0 +1,88 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: datamon-coord-pg-demo
+spec:
+  selector:
+    matchLabels:
+      app: datamon-coord-pg-demo
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: datamon-coord-pg-demo
+    spec:
+      initContainers:
+      - name: init-application-wrap
+        image: gcr.io/onec-co/datamon-pg-sidecar:latest
+        imagePullPolicy: "$PULL_POLICY"
+        command: ["sh", "-c",
+        "mkdir /tmp/coord/.scripts; cp wrap_application.sh /tmp/coord/.scripts/wrap_application.sh; chmod a+x /tmp/coord/.scripts/wrap_application.sh; echo 'placed application wrapper'"]
+        stdin: true
+        tty: true
+        volumeMounts:
+        - mountPath: /tmp/coord
+          name: container-coord
+      containers:
+      - name: demo-app
+        image: gcr.io/onec-co/datamon-pg-demo-coord-app:latest
+        imagePullPolicy: "$PULL_POLICY"
+
+        # command: ["/bin/tini"]
+        # args: ["--", "/bin/zsh"]
+
+        command: ["/tmp/coord/.scripts/wrap_application.sh"]
+        args: ["-s", "-c", "/tmp/coord", "-b", "postgres", "--",
+        "./mock_application_pg.sh",
+        "5430", "5429",
+        "$SOME_CONST", "$IS_INIT"]
+
+        stdin: true
+        tty: true
+        volumeMounts:
+        - mountPath: /tmp/coord
+          name: container-coord
+      - name: datamon-sidecar
+        image: gcr.io/onec-co/datamon-pg-sidecar:latest
+        imagePullPolicy: "$PULL_POLICY"
+
+        # command: ["/bin/tini"]
+        # args: ["--", "/bin/zsh"]
+
+        command: ["./wrap_datamon_pg.sh"]
```

Comment:
This should be made declarative and hide the implementation details within the container.

Reply:
4c1e9bf includes a serialization format that can be specified in the YAML in a declarative fashion (environment variables).